infra/docker-swarm/terraform/iam.tf

```diff
@@ -0,0 +1,50 @@
+resource "aws_iam_instance_profile" "instance_profile" {
+  name = "${var.cluster}-instance-profile"
+  role = aws_iam_role.role.name
+}
+
+resource "aws_iam_role" "role" {
+  name = "${var.cluster}-ecr-role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "policy" {
+  name        = "${var.cluster}-ecr-policy"
+  description = "A policy giving access to full access to ECR"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ecr:*"
```

Comment:
I believe all instances need to have access to AWS ECR, so that they download the docker test plan image which would be run. Also `testground` which would probably run on one of these instances needs to have access to AWS ECR to push the built images. Related to: https://github.com/ipfs/testground/pull/213

Reply:
Correct. The workers need only "list" and "download" access, i.e. read-only. The daemon needs full access. Happy to keep it on "ecr:*" for now. We can tighten the security later, but please open an issue to remember.